geopandas/tools/sjoin.py

```diff
@@ -186,6 +180,351 @@ def find_contains(a1, a2):
             columns={"_key_left": "_key_right", "_key_right": "_key_left"}
         )
 
+    # ------------------- HANDLE HOW PARAM, CREATE FINAL DF --------------------
+    return _join_results(
+        result,
+        left_df,
+        right_df,
+        how,
+        left_index_name,
+        right_index_name,
+        index_left,
+        index_right,
+        lsuffix,
+        rsuffix,
+    )
+
+
+def sjoin_nearest(
+    left_df,
+    right_df,
+    how="inner",
+    lsuffix="left",
+    rsuffix="right",
+    search_radius=None,
+    max_search_neighbors=None,
+    nearest_distances=False,
+):
+    """Spatial join of two GeoDataFrames, matching by nearest neighbor.
+    Results can be restricted to a radius using the search_radius parameter.
+    If the search is not bounded by a radius, execution can be sped up by
+      using the max_search_neighbors parameter.
+    Mixing both options usually does not improve performance over using only
+     search_radius and is not recommended.
```

Comment:
Hmm, can we find a way how to use both to get improved performance? I can imagine a task to get the nearest building but only if it is closer than 1km. Then I can easily search over thousands of buildings within the radius, while two nearest neighbours would do the same.

Reply:
The issue is that once you filter possible matches by radius, I think you'd have to rebuild the spatial index to then query the subset of possible matches by nearest. In my testing, you can still use both and it _will_ provide a performance improvement in some scenarios, but it really depends on the chosen parameters and dataset.

Reply:
That is true, didn't realise it. That would be expensive thing to do.

Reply:
Yeah, it's unfortunate. I tried :(
Can I resolve this comment and keep that part as-is?

Reply:
Keep it unresolved so someone else can see it as well, but keep this part as it is for now, unless we'll find a better way.